geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/text/Suffixes.scala

```diff
@@ -0,0 +1,83 @@
+/***********************************************************************
+* Copyright (c) 2013-2016 Commonwealth Computer Research, Inc.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Apache License, Version 2.0
+* which accompanies this distribution and is available at
+* http://www.opensource.org/licenses/apache2.0.php.
+*************************************************************************/
+
+package org.locationtech.geomesa.utils.text
+
+import java.util.regex.Pattern
+
+import com.typesafe.scalalogging.LazyLogging
+import org.joda.time.{Period, Duration}
+import org.joda.time.format.PeriodFormatterBuilder
+
+import scala.util.{Failure, Success, Try}
+
+object Suffixes {
+
+  object Time extends LazyLogging {
+    val format = new PeriodFormatterBuilder()
+      .appendDays().appendSuffix("d")
+      .appendHours().appendSuffix("h")
+      .appendMinutes().appendSuffix("m")
+      .appendSeconds().appendSuffix("s")
+      .appendMillis().appendSuffix("ms")
+      .toFormatter
+
+    def duration(s: String): Option[Duration] = {
+      Try[Duration](Period.parse(s.toLowerCase, format).toStandardDuration)
+        .orElse(Try(Duration.millis(s.toLong)))
+      match {
+        case Success(d) => Some(d)
+        case Failure(e) =>
+          logger.error(s"Unable to parse duration $s", e)
```

Comment:
this error might be slightly misleading since the first exception from the period parser is being dropped